yt/visualization/plot_container.py

```diff
@@ -80,16 +82,16 @@ def newfunc(*args, **kwargs):
 def validate_plot(f):
     @wraps(f)
     def newfunc(*args, **kwargs):
-        if hasattr(args[0], '_data_valid'):
-            if not args[0]._data_valid:
-                args[0]._recreate_frb()
-        if hasattr(args[0], '_profile_valid'):
-            if not args[0]._profile_valid:
-                args[0]._recreate_profile()
-        if not args[0]._plot_valid:
+        plot = args[0]
+        if not plot.data_is_valid:
+            plot._recreate_frb()
+        if hasattr(plot, '_profile_valid'):
+            if not plot._profile_valid:
```

Comment:
these two lines can be merged actually. The second condition is only evaluated if the first one is `True`